### PR TITLE
Added verify email page on WooCommerce landing page

### LINKED ISSUE #64165: Woo Email Confirmation - Update Woo Calypso homepage for non-verified users
If the user reaches Calypso and clicks on WooCommerce, but has not yet verified their account email, we should modify the information displayed to them to prompt them to confirm their email before they progress.

image

See DekZ88lDqG8Je04SyG1ZBK-fi-0%3A1 for full flow information.

### PR DESCRIPTION
**Pull request by nelsonec87**
#### Proposed Changes

If the user reaches Calypso and clicks on WooCommerce, but has not yet verified their account email, we should modify the information displayed to them to prompt them to confirm their email before they progress.

#### Testing Instructions

* Create a new site `/new` using a new account.
* After the site is created, go to the WooCommerce menu
* You should see the new page
image

* Click on 'Edit email address'. It should redirect you to the profile settings.
* Go back
* Click on 'Resend verification email'
* Check if you've received the email
* Verify the email
* Refresh the page
* You should see the regular page
 
image

Closes #64165

### CODE REVIEW

File: client/my-sites/woocommerce/landing-page.tsx [hunk 1]
```diff
@@ -5,15 +5,18 @@ import { sprintf, _x } from '@wordpress/i18n';
...
 import EmptyContent from 'calypso/components/empty-content';
 import FixedNavigationHeader from 'calypso/components/fixed-navigation-header';
 import FormattedHeader from 'calypso/components/formatted-header';
 import InlineSupportLink from 'calypso/components/inline-support-link';
 import PromoSection, { Props as PromoSectionProps } from 'calypso/components/promo-section';
 import WarningCard from 'calypso/components/warning-card';
+import { useSendEmailVerification } from 'calypso/landing/stepper/hooks/use-send-email-verification';
 import useScrollAboveElement from 'calypso/lib/use-scroll-above-element';
 import useWooCommerceOnPlansEligibility from 'calypso/signup/steps/woocommerce-install/hooks/use-woop-handling';
+// import { requestEmailVerification } from 'calypso/state/data-layer/wpcom/me/send-verification-email';
```
Comment (markbiek) [inline client/my-sites/woocommerce/landing-page.tsx]: Should we remove this commented-out line?

**Review by markbiek — APPROVED** [review]
One small note but otherwise LGTM!

File: client/my-sites/woocommerce/landing-page.tsx (hunk 1 above)
Comment (nelsonec87) [reply client/my-sites/woocommerce/landing-page.tsx]: Of course :)  🤦‍♂️

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7366456

Thank you @nelsonec87 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7366456).

**Pull request merged**